CrossAppLoginUI/src/main/kotlin/com/infomaniak/core/crossloginui/views/CrossLoginSelectAccountsView.kt

```diff
@@ -0,0 +1,109 @@
+import com.infomaniak.core.crossloginui.views.components.CrossLoginSelectAccounts
+
+class CrossLoginSelectAccountsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : AbstractComposeView(context, attrs, defStyleAttr) {
+
+    private val accounts = mutableStateListOf<CrossLoginUiAccount>()
+
+    private var onClickListener: (() -> Unit)? = null
+
+    private var primaryColor by mutableStateOf<Color?>(null)
+    private var onPrimaryColor by mutableStateOf<Color?>(null) // TODO
+    private var titleColor: Color? = null
+    private var descriptionColor: Color? = null
+    private var avatarStrokeColor: Color? = null
+    private var buttonStrokeColor: Color? = null
+
+    init {
+        context.obtainStyledAttributes(attrs, R.styleable.CrossLoginSelectAccountsView, defStyleAttr, 0).apply {
+            primaryColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginPrimaryColor)
+            onPrimaryColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginOnPrimaryColor)
+            titleColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginTitleColor)
+            descriptionColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginDescriptionColor)
+            avatarStrokeColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginAvatarStrokeColor)
+            buttonStrokeColor = getColorOrNull(R.styleable.CrossLoginSelectAccountsView_crossLoginButtonStrokeColor)
+            recycle()
+        }
+    }
+
+    @Composable
+    override fun Content() {
+        MaterialThemeFromXml {
+
+            val colors = CrossLoginDefaults.colors(
+                primary = primaryColor,
+                onPrimary = onPrimaryColor,
+                title = titleColor,
+                description = descriptionColor,
+                avatarStroke = avatarStrokeColor,
+                buttonStroke = buttonStrokeColor,
+            )
+
+            CrossLoginSelectAccounts(
+                accounts = { accounts },
+                colors = colors,
+                onClick = { onClickListener?.invoke() },
+            )
+        }
+    }
+
+    fun setPrimaryColor(@ColorInt newPrimaryColor: Int) {
+        primaryColor = Color(newPrimaryColor)
+    }
+
+    fun setOnPrimaryColor(@ColorInt newOnPrimaryColor: Int) {
+        onPrimaryColor = Color(newOnPrimaryColor)
+    }
+
+    fun setAccounts(list: List<CrossLoginUiAccount>) {
+        accounts.apply {
+            clear()
+            addAll(list)
+        }
+    }
+
+    fun setOnClickListener(listener: () -> Unit) {
+        onClickListener = listener
+    }
+
+    @Suppress("DeprecatedCallableAddReplaceWith")
+    @Deprecated("Use the other methods to set click listeners when a reaction is clicked and when the add reaction button is clicked")
+    override fun setOnClickListener(listener: OnClickListener?) = super.setOnClickListener(listener)
```

Comment:
Don't bother deprecating the standard click listener method, use it instead for your own click listener it will be easier

Reply:
No I meant, use the standard click listener for your need, don't define a new one